code/network/multi_fstracker.cpp

```diff
@@ -1360,6 +1366,56 @@ void multi_fs_tracker_report_stats_results()
 	}
 }
 
+// report the status of PXO game probe (firewall check)
+void multi_fs_tracker_report_probe_status(int flags, int next_try)
+{
+	static int last_flags = 0;
+	SCP_string str;
+
+	// if the flags haven't changed since last time then just bail
+	// *except* if the probe failed since we always want that message
+	// (don't & the flag check here, needs to be exact)
+	if ( (flags == last_flags) && (flags != PROBE_FLAGS::FAILURE) ) {
+		return;
+	}
+
+	if (flags & PROBE_FLAGS::PENDING) {
+		// if we've been here before just bail (to avoid log spam)
+		if (last_flags) {
+			last_flags = flags;
+			return;
+		}
+
+		str = "<PXO firewall probe in progress...>";
+	} else if (flags & PROBE_FLAGS::FAILURE) {
+		char t_str[64];
+
+		str = "<PXO firewall probe failed! Next attempt in ";
+
+		if (next_try < 120) {
+			SDL_snprintf(t_str, SDL_arraysize(t_str), "%d seconds...>", next_try);
+		} else if (next_try < (60*60+1)) {
+			int minutes = next_try / 60;
+			SDL_snprintf(t_str, SDL_arraysize(t_str), "%d minutes...>", minutes);
+		} else {
+			int hours = next_try / (60*60);
+			SDL_snprintf(t_str, SDL_arraysize(t_str), "%d hours...>", hours);
+		}
+
+		str += t_str;
+	} else if (flags & PROBE_FLAGS::SUCCESS) {
+		str = "<PXO firewall probe was a success!>";
+	} else {
+		// getting here shouldn't happen, but it's not technically fatal
+		return;
```

Comment:
Maybe a one-time mprintf() or multilog line with the value that caused it to reach here would be helpful to catch hard to  find bugs?

Reply:
There could technically be a case for that to happen which is why I handled it. But it's also self-correcting so there really isn't a need to be notified of it.